build: turn off the most of feature flags on Windows platform

fujimotos opened the pull request:
This is a temporal patch that allows us to compile the source tree
on Windows. Our plan is to next migrate each plugin one by one and
then enables ones that get done.

Part of #960

edsiper commented:
what do you think about adding some kind of cmake/windows-setup.cmake that can be invoked to avoid having all options in the main CMakeListst.txt ?

fujimotos commented:
> what do you think about adding some kind of cmake/windows-setup.cmake that can be invoked to avoid having all options in the main CMakeListst.txt ?

OK. I moved the custom settings to a separate file in 9106dc1,
and also added comments to make review a little bit easier.
